src/storage/raw/raw_mvcc.rs

```diff
@@ -28,8 +28,9 @@ impl<S: Snapshot> RawMvccSnapshot<S> {
         iter_opt.set_fill_cache(opts.map_or(true, |v| v.fill_cache()));
         iter_opt.use_prefix_seek();
         iter_opt.set_prefix_same_as_start(true);
-        let end_key = key.clone().append_ts(TimeStamp::zero());
-        iter_opt.set_upper_bound(end_key.as_encoded(), DATA_KEY_PREFIX_LEN);
+        let mut end_key = key.clone().append_ts(TimeStamp::zero()).into_encoded();
+        end_key.push(0); // `end_key` is inclusive.
```

Comment:
I don't think it's safe. For write cf, the prefix is determined by subtracting a fixed suffix. And I don't understand why `end_key` is inclusive. We always use `[start, end)`, anything breaks this convention will run into chaos.

Reply:
You miss my point. Whether it's safe depends on how prefix is extracted. For extractor that takes a fixed suffix, appending a `0` is wrong.

I think asserting Timestamp can't be zero is a reasonable fix.

Reply:
OK, I see. Thanks for your patience~
The issue is fixed and some assertions are added. PTAL~